include/index_config.h

```diff
@@ -0,0 +1,193 @@
+#include "common_includes.h"
+#include "parameters.h"
+
+namespace diskann
+{
+enum DataStoreStrategy
+{
+    MEMORY
+};
+
+enum GraphStoreStrategy
+{
+};
+struct IndexConfig
+{
+    DataStoreStrategy data_strategy;
+    GraphStoreStrategy graph_strategy;
+
+    Metric metric;
+    size_t dimension;
+    size_t max_points;
+
+    bool dynamic_index;
+    bool enable_tags;
+    bool pq_dist_build;
+    bool concurrent_consolidate;
+    bool use_opq;
+
+    size_t num_pq_chunks;
+    size_t num_frozen_pts;
+
+    std::string label_type = "uint32";
```

Comment:
Do we need this and the next initializer given that the only constructor always sets them?